src/nitpick/style.py

```diff
@@ -733,6 +749,158 @@ def _download(self, url: furl, **kwargs) -> str:
         return super()._download(github_url.raw_content_url, **kwargs)
 
 
+@dataclass(frozen=True)
+class GitLabURL:
+    """Represent a GitLab URL, created from a URL or from its parts."""
+
+    scheme: str
+    host: str
+    project: list[str]
+    path: str
+    git_reference: str
+    query_params: tuple[tuple[str, str], ...]
+    auth_token: str | None = None
+
+    @property
+    def token(self) -> str | None:
+        """Token encoded in this URL.
+
+        If present, and it starts with a ``$``, it will be replaced with the
+        value of the environment corresponding to the remaining part of the
+        string.
+        """
+        token = self.auth_token
+        if token is not None and token.startswith("$"):
+            token = os.getenv(token[1:])
```

Comment:
Maybe this works, I haven't tried.
```suggestion
        if not token and token.startswith("$"):
            token = os.getenv(token[1:])
```